controller/network/route_perf_test.go

```diff
@@ -0,0 +1,173 @@
+		{cost: 240, path: []string{"router-000", "router-009"}},
+		{cost: 213, path: []string{"router-000", "router-002", "router-010"}},
+		{cost: 198, path: []string{"router-000", "router-050", "router-011"}},
+		{cost: 187, path: []string{"router-000", "router-035", "router-012"}},
+		{cost: 204, path: []string{"router-000", "router-002", "router-013"}},
+		{cost: 153, path: []string{"router-000", "router-014"}},
+		{cost: 306, path: []string{"router-000", "router-051", "router-015"}},
+		{cost: 253, path: []string{"router-000", "router-056", "router-016"}},
+		{cost: 232, path: []string{"router-000", "router-050", "router-011", "router-017"}},
+		{cost: 226, path: []string{"router-000", "router-020", "router-018"}},
+		{cost: 119, path: []string{"router-000", "router-019"}},
+		{cost: 188, path: []string{"router-000", "router-020"}},
+		{cost: 268, path: []string{"router-000", "router-002", "router-021"}},
+		{cost: 239, path: []string{"router-000", "router-051", "router-022"}},
+		{cost: 118, path: []string{"router-000", "router-023"}},
+		{cost: 190, path: []string{"router-000", "router-002", "router-024"}},
+		{cost: 137, path: []string{"router-000", "router-025"}},
+		{cost: 252, path: []string{"router-000", "router-023", "router-073"}},
+		{cost: 237, path: []string{"router-000", "router-002", "router-054", "router-072"}},
+		{cost: 218, path: []string{"router-000", "router-051", "router-071"}},
+		{cost: 197, path: []string{"router-000", "router-002", "router-070"}},
+		{cost: 254, path: []string{"router-000", "router-050", "router-069"}},
+		{cost: 238, path: []string{"router-000", "router-019", "router-068"}},
+		{cost: 197, path: []string{"router-000", "router-002", "router-067"}},
+		{cost: 290, path: []string{"router-000", "router-050", "router-066"}},
+		{cost: 230, path: []string{"router-000", "router-051", "router-065"}},
+		{cost: 283, path: []string{"router-000", "router-002", "router-054", "router-064"}},
+		{cost: 320, path: []string{"router-000", "router-063"}},
+		{cost: 248, path: []string{"router-000", "router-002", "router-062"}},
+		{cost: 329, path: []string{"router-000", "router-002", "router-061"}},
+		{cost: 280, path: []string{"router-000", "router-056", "router-060"}},
+		{cost: 257, path: []string{"router-000", "router-084", "router-059"}},
+		{cost: 231, path: []string{"router-000", "router-084", "router-058"}},
+		{cost: 211, path: []string{"router-000", "router-002", "router-054", "router-057"}},
+		{cost: 155, path: []string{"router-000", "router-056"}},
+		{cost: 280, path: []string{"router-000", "router-002", "router-055"}},
+		{cost: 198, path: []string{"router-000", "router-002", "router-054"}},
+		{cost: 308, path: []string{"router-000", "router-002", "router-053"}},
+		{cost: 273, path: []string{"router-000", "router-052"}},
+		{cost: 133, path: []string{"router-000", "router-051"}},
+		{cost: 141, path: []string{"router-000", "router-050"}},
+	}
+
+	b.StartTimer()
```

Comment:
just a comment: maybe StartTimer should "stop" the timer too? i keep seeing the 'StopTimer' function call at the top of these tests which I don't quite 'get why' other than just making sure that the setup bits aren't swept into the timer?

Reply:
It's just how go benchmark stuff works. By default the whole function is timed, but you can stop the timer and restart when you're done with expensive initialization.